### PR TITLE
BHBC-1555_2

### PR DESCRIPTION
**Pull request by cgarrettjones**
# Overview

## Links to jira tickets

- [BHBC-1555](https://quartech.atlassian.net/browse/BHBC-1555)

## This PR contains the following changes

- data model enhancement to support treatments

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: database/src/migrations/release.0.2/restoration.sql [hunk 1]
```diff
@@ -459,8 +503,52 @@ COMMENT ON COLUMN iucn_conservation_action_level_3_subclassification.revision_co
...
+    revision_count            integer           DEFAULT 0 NOT NULL,
+    CONSTRAINT linear_feature_type_pk_1 PRIMARY KEY (linear_feature_type_id)
+)
+;
+
+
+
+COMMENT ON COLUMN linear_feature_type.linear_feature_type_id IS 'System generated surrogate primary key identifier.'
+;
+COMMENT ON COLUMN linear_feature_type.name IS 'Name of the First Nation.'
```
Comment (NickPhura) [inline database/src/migrations/release.0.2/restoration.sql]: `First Nation`

File: database/src/migrations/release.0.2/restoration.sql [hunk 2]
```diff
@@ -955,12 +1043,12 @@ COMMENT ON COLUMN project_spatial_component_type.project_spatial_component_type_
 ;
 COMMENT ON COLUMN project_spatial_component_type.name IS 'Name of the First Nation.'
```
Comment (NickPhura) [inline database/src/migrations/release.0.2/restoration.sql]: also First Nation

File: database/src/migrations/release.0.2/restoration.sql [hunk 3]
```diff
@@ -1370,8 +1458,243 @@ COMMENT ON COLUMN system_user_role.revision_count IS 'Revision count used for co
...
+    revision_count           integer           DEFAULT 0 NOT NULL,
+    CONSTRAINT treatment_type_pk PRIMARY KEY (treatment_type_id)
+)
+;
+
+
+
+COMMENT ON COLUMN treatment_type.treatment_type_id IS 'System generated surrogate primary key identifier.'
+;
+COMMENT ON COLUMN treatment_type.name IS 'Name of the First Nation.'
```
Comment (NickPhura) [inline database/src/migrations/release.0.2/restoration.sql]: first nation

File: database/src/migrations/release.0.2/restoration.sql [hunk 4]
```diff
@@ -1370,8 +1458,243 @@ COMMENT ON COLUMN system_user_role.revision_count IS 'Revision count used for co
...
+    revision_count                              integer           DEFAULT 0 NOT NULL,
+    CONSTRAINT treatment_unit_spatial_component_type_pk PRIMARY KEY (treatment_unit_spatial_component_type_id)
+)
+;
+
+
+
+COMMENT ON COLUMN treatment_unit_spatial_component_type.treatment_unit_spatial_component_type_id IS 'System generated surrogate primary key identifier.'
+;
+COMMENT ON COLUMN treatment_unit_spatial_component_type.name IS 'Name of the First Nation.'
```
Comment (NickPhura) [inline database/src/migrations/release.0.2/restoration.sql]: first nation

File: database/src/migrations/release.0.2/restoration.sql [hunk 5]
```diff
@@ -1461,566 +1784,705 @@ COMMENT ON COLUMN webform_draft.revision_count IS 'Revision count used for concu
...
-CREATE INDEX "Ref128119" ON project(project_type_id)
+CREATE INDEX "Ref3023" ON project(project_type_id)
 ;
---
--- INDEX: project_report_attachment_uk1
+-- 
+-- INDEX: project_report_attachment_uk1 
 --
 
 CREATE UNIQUE INDEX project_report_attachment_uk1 ON project_attachment(project_id, file_name)
```
Comment (NickPhura) [inline database/src/migrations/release.0.2/restoration.sql]: Maybe it doesnt matter, but should this have `report` in the name?

File: database/src/migrations/release.0.2/restoration.sql (hunk 4 above)
Comment (cgarrettjones) [reply database/src/migrations/release.0.2/restoration.sql]: whats with the first nation?! will adjust...

File: database/src/migrations/release.0.2/restoration.sql (hunk 5 above)
Comment (cgarrettjones) [reply database/src/migrations/release.0.2/restoration.sql]: yep, will adjust.

**Comment by anissa-agahchen** [comment]
Found this issue when creating a permit
Screen Shot 2022-01-19 at 1 47 11 PM

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**